File: src/main/java/com/busted_moments/core/config/ConfigScreen.java
package com.busted_moments.core.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

import static com.busted_moments.client.Client.CONFIG;

public class ConfigScreen implements ModMenuApi {
   @Override
   public ConfigScreenFactory<?> getModConfigScreenFactory() {
      return parent -> CONFIG.build(parent).build();
   }
}

File: src/main/kotlin/com/busted_moments/client/framework/config/ConfigScreen.kt
package com.busted_moments.client.framework.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.minecraft.client.gui.screens.Screen

object ConfigScreen : ModMenuApi, ConfigScreenFactory<Screen> {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> =
        this

    override fun create(screen: Screen?): Screen =
        Config.open(screen).build()
}